services/uploads.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+var v = require('./service-helpers/validator');
+var createServiceFactory = require('./service-helpers/create-service-factory');
+
+/**
+ * Uploads API service.
+ */
+var Uploads = {};
+
+/**
+ * List all recent upload statuses
+ *
+ * See the [public documentation](https://www.mapbox.com/api-documentation/#uploads).
+ *
+ * @param {Object} [config]
+ * @param {boolean} [config.reverse] - List uploads in chronological order, rather than reverse chronological order.
+ * @return {MapiRequest}
+ */
+Uploads.listUploads = function(config) {
+  v.assertShape({
+    reverse: v.boolean
+  })(config);
+
+  return this.client.createRequest({
+    method: 'GET',
+    path: '/uploads/v1/:ownerId',
+    query: config
+  });
+};
+
+/**
+ * Create s3 credentials.
+ *
+ * See the [public documentation](https://www.mapbox.com/api-documentation/#retrieve-s3-credentials).
+ *
+ * @param {Object} [config]
+ * @return {MapiRequest}
+ */
+Uploads.createUploadCredentials = function(config) {
+  v.assertShape({})(config);
+
+  return this.client.createRequest({
+    method: 'POST',
+    path: '/uploads/v1/:ownerId/credentials'
+  });
+};
+
+/**
+ * Create an upload.
+ *
+ * See the [public documentation](https://www.mapbox.com/api-documentation/#create-an-upload).
+ *
+ * @param {Object} [config
+ * @param {string} config.tileset - the map ID to create or replace in the format  username.nameoftileset - limited to 32 characters (only  - and  _ special characters allowed, limit does not include username)
```

Comment:
Let's call this `mapId` — I think that pretty consistently should refer to the `{ownerId}.{tilesetId}` pairing, whereas `tileset` is kind of ambiguous.